File: 1512. Number of Good pairs/GoodPairs/src/GoodPairs.java
public class GoodPairs {
    public static void main(String[] args) {
        int [] nums = {1,2,3};
        System.out.println(GoodPairs.goodPairsSolution(nums));
    }

    public static int goodPairsSolution(int [] nums) {
        int pairs = 0;
        for (int i = 0; i < nums.length; i++) {
            for (int j = 0; j < nums.length; j++) {
                if (nums[i] == nums[j] && i < j) pairs++;
            }
        }

        return pairs;
    }
}

File: 1512. Number of Good pairs/GoodPairsCsharp/GoodPairs.cs
namespace GoodPairsCsharp;

public class GoodPairs
{
    public static void Main(string[] args)
    {
        var nums = new [] {1,2,3};
        Console.WriteLine(NumIdenticalPairs(nums));
    }

    private static int NumIdenticalPairs(int [] nums)
    {
        var pairs = 0;
        int i = 0, j = 0;

        while (i < nums.Length)
        {
            if (i < j && nums[i] == nums[j]) pairs++;
            j++;
            if (j != nums.Length) continue;
            i++;
            j = 0;
        }

        return pairs;
    }
}